internal/builders/maven/action.yml

```diff
@@ -68,12 +68,39 @@ runs:
       shell: bash
       env:
         SLSA_OUTPUTS_ARTIFACTS_FILE: ${{ inputs.slsa-layout-file }}
+        PROJECT_ROOT: ${{ fromJson(inputs.slsa-workflow-inputs).directory }}
```

Comment:
It's usually a good idea to put an "UNTRUSTED" prefix on user inputs just to make it clear.
```suggestion
        UNTRUSTED_PROJECT_ROOT: ${{ fromJson(inputs.slsa-workflow-inputs).directory }}
```

Then after validation you should explicitly set it to a trusted value.
```shell
validate_path "${UNTRUSTED_PROJECT_ROOT}"

# Directory was validated. Explicitly trust it.
project_root="${UNTRUSTED_PROJECT_ROOT}"
```

Reply:
Done.